nano/core_test/node.cpp

```diff
@@ -1521,6 +1521,33 @@ TEST (node, rep_list)
 	}
 }
 
+TEST (node, rep_weight)
+{
+	nano::system system (24000, 1);
+	auto & node (*system.nodes[0]);
+
+	node.peers.insert (nano::endpoint (boost::asio::ip::address_v6::loopback (), 24001), 0);
+	ASSERT_TRUE (node.rep_crawler.representatives (1).empty ());
+	nano::endpoint endpoint0 (boost::asio::ip::address_v6::loopback (), 24000);
+	nano::endpoint endpoint1 (boost::asio::ip::address_v6::loopback (), 24002);
+	nano::endpoint endpoint2 (boost::asio::ip::address_v6::loopback (), 24003);
+	nano::amount amount100 (100);
+	nano::amount amount50 (50);
+	node.peers.insert (endpoint2, nano::protocol_version);
+	node.peers.insert (endpoint0, nano::protocol_version);
+	node.peers.insert (endpoint1, nano::protocol_version);
+	nano::keypair keypair1;
+	nano::keypair keypair2;
+	node.rep_crawler.response (endpoint0, keypair1.pub, amount100);
+	node.rep_crawler.response (endpoint1, keypair2.pub, amount50);
+	// Make sure we get the rep with the most weight first
+	auto reps (node.rep_crawler.representatives (1));
```

Comment:
Could we also check total representatives size == 2 ?

Reply:
Sure. The rep container isn't exposed, but I could add a total_size function for tests.